benchmark/utils/nodeRedUtils.ts

```diff
@@ -0,0 +1,144 @@
+import http from 'http';
+import * as dotenv from 'dotenv';
+dotenv.config();
+
+const NODE_RED_PORT: number = 1880;
+const NODE_RED_USERNAME = 'admin';
+const NODE_RED_PASSWORD = 'admin';
+
+
+function generateAuthHeader(username?: string, password?: string): string | undefined {
+  if (username && password) {
+    const credentials = `${username}:${password}`;
+    const encodedCredentials = Buffer.from(credentials).toString('base64');
+    return `Basic ${encodedCredentials}`;
+  }
+  return undefined;
+}
+
+async function checkNodeRedStatus(): Promise<boolean> {
+  return new Promise((resolve) => {
+    const req = http.request({
+      hostname: 'localhost',
```

Comment:
This is a bit nitpick, but I prefer that we use the real address (instead of localhost). With the IP:

* We avoid an extra DNS resolution step
* We avoid misconfiguration in the system's hosts file
* It's straightforward and will always point to the resource (loopback address) we are looking for.

```suggestion
      hostname: '127.0.0.1',
```

Change this in all the other places where you have been using localhost in this PR please.

Reply:
Fixed too